### DISCUSSION TITLE
Github Uploads

### DISCUSSION THREAD

**Discussion by SHARON-CHATTOPADHYAY** [drknzz/GitHub-Achievements#314] [Ideas]
How to upload whole project in the github in one go?

**Comment by Anonymous15052003** [comment]
So, for uploading all files and folders, you need to use github command line terminal in VS Code and initiallise the git into the vs terminal and commit all the changes into the remote repo using git commands and push it finally.

**Comment by ghost** [comment]
To upload a whole project, you need to make one and I assume you must copy the contents of it to that project.

**Comment by Limzen** [comment]
To upload a whole project to GitHub in one go, follow these steps:

### Step 1: Initialize Git in Your Project Folder
1. Open a terminal or command prompt.
2. Navigate to your project directory:
   ```bash
   cd path/to/your/project
   ```
3. Initialize a Git repository:
   ```bash
   git init
   ```

### Step 2: Add Your Files
Add all files in the directory to the Git staging area:
```bash
git add .
```

### Step 3: Commit the Changes
Commit the added files with a descriptive message:
```bash
git commit -m "Initial commit"
```

### Step 4: Create a New Repository on GitHub
1. Go to [GitHub](https://github.com) and sign in.
2. Click on **New Repository**.
3. Name your repository and keep it public or private as per your preference, then click **Create Repository**.

### Step 5: Link Your Local Repository to GitHub
Copy the URL of the GitHub repository you just created. In the terminal, link your local repository to the remote one:
```bash
git remote add origin https://github.com/your-username/your-repository.git
```

### Step 6: Push Your Project to GitHub
Push all commits to the main branch on GitHub:
```bash
git push -u origin main
```

Now your entire project is uploaded to GitHub in one go. You can visit your repository to verify the files are there.

### Notes
- If your repository uses a different branch name (e.g., `master`), replace `main` with the appropriate branch name.
- Make sure to replace `your-username` and `your-repository` with your actual GitHub username and repository name.
